test/blackbox/run_blackbox_tests/test_cases_get_node_reachability.c

```diff
@@ -203,7 +215,7 @@ static bool test_get_node_reachability_01(void) {
 	// Cleanup
 
 	meshlink_close(mesh);
-	assert(meshlink_destroy(NUT_CONF));
+	assert(meshlink_destroy(NUT_CONF_01));
```

Comment:
You could remove the calls to `meshlink_destroy()` at the end of a test if you have one at the start. The benefit is that you can then also have the possibility to inspect the generated files after the test has run.

Reply:
I believe that we should neither use `meshlink_destroy()` at the start in our test vector, I observed that when a test case fails and we re-run the same test again it fails to destroy the confbase if placed at the start of the test (due to locking) and also we cannot assert on the destroy call as it may or may not have confbase all the time.
May be can call `system("rm -rf " NUT_CONF_01)` at the start of the test. Do you have any other ideas?

Reply:
If it fails due to locking, it means there is still an instance of MeshLink running, and that is a problem. Calling `rm -rf` might work, but that instance that is still running might write back host config files a little later. Also, if `meshlink_destroy()` fails to lock, then `meshlink_open()` would fail as well.
There are two solutions to the locking problem: either have every test in a given binary use a different config directory, so that each test can run without interfering with others, or use a teardown function to make sure all MeshLink handles are closed, even if a test fails. I think the change to use `NUT_CONF_01`, `NUT_CONF_02` etc. already does the former, so you should not see this issue anymore.

Reply:
Yeah, even the man page for flock says that `the lock is released either by an explicit  LOCK_UN  operation on any of these duplicate file descriptors, or when all such file descriptors have been closed.` but not sure why it failed to lock. Anyhow, I modified with one of your solutions by creating a unique config directory for each test run. I will let you if something is wrong over there.
One more thing that even though you call `meshlink_destroy()` it actually removes all sub-directories in the confbase and all the config files inside them, but it missed to remove main confbase or root confbase directory (in the source code I couldn't see code that removes confbase directory).

Reply:
I think it failed because CMocka does not start a new process for each test. So if one test fails an `assert_int_equals()` call, or any other assert function provided by CMocka, it aborts just that test, but then CMocka will try to start the next test.

The reason the confbase directory itself isn't destroyed is so you can have MeshLink have its configuration in a directory that itself cannot be deleted, for example in the root directory of a partition, or on Android in the application's private data directory.